src/language-js/printer-estree.js

```diff
@@ -4107,11 +4107,11 @@ function printMemberChain(path, options, print) {
   //     .map(x => x)
   //
   // In order to detect those cases, we use an heuristic: if the first
-  // node is just an identifier with the name starting with a capital
-  // letter, just a sequence of _$ or this. The rationale is that they are
-  // likely to be factories.
+  // node is an identifier with the name starting with a capital letter,
+  // containing only _ and $, or shorter than tabWidth. The rationale
+  // is that they are likely to be factories.
   function isFactory(name) {
```

Comment:
I don't think short names mean they are likely to be factories.

It's a different rationale.. so I'd have a check for `isFactory(name) || name.length <= options.tabWidth`

Reply:
That'll be a little gross to call: 

```
isFactory(groups[0][groups[0].length - 1].node.property.name) || 
  groups[0][groups[0].length - 1].node.property.name.length <= options.tabWidth
```

What about renaming the function? `isFactoryOrShort` or `isNoWrap`?

Reply:
Sure, that works

Reply:
I updated the function name and comment. The travis build is failing now though, because of formatting. Both `yarn link` and `yarn lint --fix` just hang when I try to run them locally.

Reply:
It looks like you have some trailing whitespace in a comment. Removing that should appease the linter.

Reply:
Ah. Took it out.